### ISSUE TITLE
Work together re: integration in gitlab?

### ISSUE THREAD

**Issue by elbosso** [yuzutech/kroki#328]
Hi, i just released a solution to leverage Gitlabs PlantUML integration for more visualization types likt TeX and Gnuplot (https://github.com/elbosso/plantumlinterfaceproxy) - maybe we could work together to "leverage synergies" (oh god, what am I writing...) and make Gitlab even better?

**Comment by Mogztter** [comment]
Hi Jürgen,

It seems that we had the same idea, extending the PlantUML server (API) to other diagram libraries. I already planned to add support for [WireViz](https://github.com/yuzutech/kroki/issues/282) and [Gnuplot](https://github.com/yuzutech/kroki/issues/53).

For WireViz, the integration should be straightforward. For Gnuplot, it's a bit more complicated because you can execute arbitrary commands so it's really unsafe to execute untrusted Gnuplot definitions.

> maybe we could work together to "leverage synergies" (oh god, what am I writing...) and make Gitlab even better?

Sure that sounds like a good idea 👍 
I intent to create a "formal" issue in GitLab to add built-in integration with Kroki. While working on adding Mermaid support to AsciiDoc in GitLab, I briefly described Kroki in this issue: https://gitlab.com/gitlab-org/gitlab/-/issues/31818#note_374399536

Even if Kroki is compatible with the PlantUML API, I think a built-in integration would be better.

If you want to add WireViz support in Kroki that would be great!
Currently, we have a Docker image based on Python named "yuzutech/kroki-blockdiag" that contains: BlockDiag, SeqDiag, ActDiag, NwDiag, PacketDiag and RackDiag.
I think we should create a new Docker image for WireViz. In the future, we might decide to merge them but for now I think it's better to keep them separated.

Just to give you a heads-up, I will be on vacation in August, so the project will probably move slower during this time 🌴🏖️

Is that what you had in mind?

**Comment by elbosso** [comment]
Hi,

I already created an Issue for integrating Gnuplot into Gitlab as a first class citizen some time ago (https://gitlab.com/gitlab-org/gitlab/-/issues/27541) but it never really got any attention so I closed it when the first version of my solution was released. The WireViz integration actually was what started that project and it is available als Github-Project too:  It implements a small docker-microservice offering get and Post-Requests ans a Swagger UI (https://github.com/elbosso/WireViz).

And you are right: There are security concerns with Gnuplut for the reasons you pointed out but even more so in the case of TeX (deu: https://elbosso.github.io/tex_virus.html#content; eng: https://elbosso.github.io/resources/tex-login.pdf) as there is the famous TeX-viruses. I will point that out in the README - my integration really is more for self-hosted solutions where one can be sure of all the participants are known and show at least a modicum of good-will.

As for your holidays - enjoy it!!
